test/integration/master/kube_apiserver_test.go

```diff
@@ -85,6 +85,43 @@ func TestRun(t *testing.T) {
 	}
 }
 
+func TestReadyz(t *testing.T) {
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--post-start-hook-grace-period", "10s"}, framework.SharedEtcd())
+	defer server.TearDownFn()
+	defer time.Sleep(10 * time.Second)
```

Comment:
why that sleep?

Reply:
It's removed. There were some clean-up issues in previous iterations of this test.